workflow/type.go

```diff
@@ -15,8 +25,46 @@ type task struct {
 	TaskKey      string
 }
 
+type filter struct {
+	Key       string
+	Predicate predicate
+	Value     interface{}
+}
+
 // Trigger is an event that triggers a workflow
 type trigger struct {
```

Comment:
rename trigger to hum EventTrigger?

Reply:
will be moved in the service package so let's discuss that at this time